src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -40,8 +46,31 @@ case class AvroRelation(location: String)(@transient val sqlContext: SQLContext)
     convertedSchema
   }
 
+  def convertToSparkSQL(obj: Any): Any = {
+    obj match {
+      case u: Utf8 => u.toString
+      case m: Map[Any, Any] => m.map(x => (x._1.toString, convertToSparkSQL(x._2)))
```

Comment:
Maybe add a comment to this method on the special-casing that's going on. Namely,
- Maps always have String keys
- Byte arrays and ByteBuffers are reused by Avro, so we must copy them out.